internal/version/version.go

```diff
@@ -0,0 +1,177 @@
+	Release    bool   `json:"release,omitempty"`
+}
+
+// SemVer represents a semantic version
+type SemVer struct {
+	Major int64 `json:"major"`
+	Minor int64 `json:"minor"`
+	Patch int64 `json:"patch"`
+}
+
+// AllowedVersions provides a list of SemVer per Service
+type AllowedVersions struct {
+	Bootstrap   []SemVer
+	Satellite   []SemVer
+	Storagenode []SemVer
+	Uplink      []SemVer
+	Gateway     []SemVer
+}
+
+// SemVerRegex is the regular expression used to parse a semantic version.
+// https://github.com/Masterminds/semver/blob/master/LICENSE.txt
+const SemVerRegex string = `v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?` +
+	`(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?` +
+	`(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?`
+
+// NewSemVer parses a given version and returns an instance of SemVer or
+// an error if unable to parse the version.
+func NewSemVer(regex *regexp.Regexp, v string) (*SemVer, error) {
+	m := regex.FindStringSubmatch(v)
+	if m == nil {
+		return nil, errors.New("invalid semantic version for build")
+	}
+
+	sv := SemVer{}
+
+	var err error
+
+	// first entry of m is the entire version string
+	sv.Major, err = parseToInt64(m[1])
+	if err != nil {
+		return nil, err
+	}
+
+	if m[2] == "" {
+		sv.Minor = 0
+	} else {
+		sv.Minor, err = parseToInt64(m[2])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if m[3] == "" {
+		sv.Patch = 0
+	} else {
+		sv.Patch, err = parseToInt64(m[3])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &sv, nil
+}
+
+// String converts the SemVer struct to a more easy to handle string
+func (sem *SemVer) String() (version string) {
+	return fmt.Sprintf("v%d.%d.%d", sem.Major, sem.Minor, sem.Patch)
+}
+
+// New creates Version_Info from a json byte array
+func New(data []byte) (v Info, err error) {
+	err = json.Unmarshal(data, &v)
+	return v, err
+}
+
+// Marshal converts the existing Version Info to any json byte array
+func (v Info) Marshal() (data []byte, err error) {
+	data, err = json.Marshal(v)
+	return
+}
+
+// parseToInt64 converts a string with schema .xxx to an int64 or returns an error
+func parseToInt64(label string) (int64, error) {
+	tmp := strings.TrimPrefix(label, ".")
+	l, err := strconv.ParseInt(tmp, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("invalid semantic version: %s", err)
+	}
+
+	return l, nil
+}
+
+// containsVersion compares the allowed version array against the passed version
+func containsVersion(all []SemVer, x SemVer) bool {
+	for _, n := range all {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+// StrToSemVerList converts a list of versions to a list of SemVer
+func StrToSemVerList(serviceVersions []string) (versions []SemVer, err error) {
+
+	versionRegex := regexp.MustCompile("^" + SemVerRegex + "$")
+
+	for _, subversion := range serviceVersions {
+		sVer, err := NewSemVer(versionRegex, subversion)
+		if err != nil {
+			return nil, err
+		}
+		versions = append(versions, *sVer)
+	}
+	return versions, err
+}
+
+func init() {
+	if Version == "" {
```

Comment:
we should not return if any of the linker flag values are set. we should only return if all of them are unset

Reply:
This cannot be done. I changed it to now set a default version of v0.0.1
Else Storj-Sim is unable to run unless we let it build binaries always with these flags.